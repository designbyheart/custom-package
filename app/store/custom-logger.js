//@flow
import RNFetchBlob from 'react-native-fetch-blob'
import uniqueId from 'react-native-unique-id'
import {
  setVcxLogger,
  writeToVcxLog,
  encryptVcxLog,
} from '../bridge/react-native-cxs/RNCxs'
import type { Dispatch } from 'redux'
import { UPDATE_LOG_ISENCRYPTED } from '../send-logs/type-send-logs'

export const CUSTOM_LOG_UTILS = {
  encryptionKey: '',
}

export const customLogger = {
  recordBuffer: [],
  vcxLogFile: null,
  encryptedLogFile: null,
  initOnlyOnce: false,
  logLevel: null,
  alsoLogToConsole: process.env.NODE_ENV !== 'test' && __DEV__,
  MAX_ALLOWED_FILE_BYTES: 10000000,
  log: function(...allArgs: any[]) {
    //console.log(JSON.stringify(allArgs))
    this.addRecord({ levelName: 'log', args: allArgs })
    if (this.alsoLogToConsole) {
      console.log.apply(null, allArgs)
    }
  },
  error: function(...allArgs: any[]) {
    this.addRecord({ levelName: 'error', args: allArgs })
    if (this.alsoLogToConsole) {
      console.error.apply(null, allArgs)
    }
  },
  assert: function(...allArgs: any[]) {
    if (this.alsoLogToConsole) {
      console.assert.apply(null, allArgs)
    }
  },
  clear: function(...allArgs: any[]) {
    if (this.alsoLogToConsole) {
      console.clear.apply(null, allArgs)
    }
  },
  count: function(...allArgs: any[]) {
    if (this.alsoLogToConsole) {
      console.count.apply(null, allArgs)
    }
  },
  debug: function(...allArgs: any[]) {
    this.addRecord({ levelName: 'debug', args: allArgs })
    if (this.alsoLogToConsole) {
      console.debug.apply(null, allArgs)
    }
  },
  dir: function(...allArgs: any[]) {
    if (this.alsoLogToConsole) {
      console.dir.apply(null, allArgs)
    }
  },
  dirxml: function(...allArgs: any[]) {
    if (this.alsoLogToConsole) {
      console.dirxml.apply(null, allArgs)
    }
  },
  group: function(...allArgs: any[]) {
    if (this.alsoLogToConsole) {
      console.group.apply(null, allArgs)
    }
  },
  groupCollapsed: function(...allArgs: any[]) {
    if (this.alsoLogToConsole) {
      console.groupCollapsed.apply(null, allArgs)
    }
  },
  groupEnd: function(...allArgs: any[]) {
    if (this.alsoLogToConsole) {
      console.groupEnd.apply(null, allArgs)
    }
  },
  info: function(...allArgs: any[]) {
    if (this.alsoLogToConsole) {
      console.info.apply(null, allArgs)
    }
  },
  table: function(...allArgs: any[]) {
    if (this.alsoLogToConsole) {
      console.table.apply(null, allArgs)
    }
  },
  time: function(...allArgs: any[]) {
    if (this.alsoLogToConsole) {
      console.time.apply(null, allArgs)
    }
  },
  timeEnd: function(...allArgs: any[]) {
    if (this.alsoLogToConsole) {
      console.timeEnd.apply(null, allArgs)
    }
  },
  trace: function(...allArgs: any[]) {
    this.addRecord({ levelName: 'trace', args: allArgs })
    if (this.alsoLogToConsole) {
      console.trace.apply(null, allArgs)
    }
  },
  warn: function(...allArgs: any[]) {
    this.addRecord({ levelName: 'warn', args: allArgs })
    if (this.alsoLogToConsole) {
      console.warn.apply(null, allArgs)
    }
  },

  captureException: function(error: Error) {
    this.log(error.toString())
  },

  setVcxLogFile: function(logFile: string) {
    this.vcxLogFile = logFile
  },

  getVcxLogFile: function() {
    return this.vcxLogFile
  },

  getEncryptedVcxLogFile: function() {
    return this.encryptedLogFile
  },

  init: function(levelName: string) {
    if (!this.initOnlyOnce) {
      this.initOnlyOnce = true
      this.logLevel = levelName

      const fetchPromise = fetch(
        'https://connect.me/sendlogs.public.encryption.key.txt'
      )
      if (fetchPromise) {
        fetchPromise
          .then(function(response) {
            return response.text()
          })
          .then(function(verKey) {
            //console.log('The encryption key is: ', verKey)
            CUSTOM_LOG_UTILS.encryptionKey = verKey
          })
      }

      // invoke the RNIndy.setVcxLogger function
      uniqueId()
        .then(uniqueIdent => {
          //console.log('The app unique id is: ', uniqueIdent)
          setVcxLogger(this.logLevel, uniqueIdent, this.MAX_ALLOWED_FILE_BYTES)
            .then(logFilePath => {
              this.setVcxLogFile(logFilePath)
              //console.log('Setting vcx log file to: ', logFilePath)
            })
            .catch(error => {
              console.error('Error setting vcx log file: ', error)
            })
        })
        .catch(error => {
          console.error('Error getting unique id in redux store: ', error)
        })
    }
  },

  addRecord: function(record: any) {
    // write the logging record to the debug file
    const rotatingLog = this.getVcxLogFile()
    if (rotatingLog) {
      RNFetchBlob.fs
        .exists(rotatingLog)
        .then(exists => {
          if (exists) {
            if (this.recordBuffer.length) {
              for (let i = 0; i < this.recordBuffer.length; ++i) {
                this.writeToLog(rotatingLog, this.recordBuffer[i])
              }
              this.recordBuffer.length = 0
            }
            this.writeToLog(rotatingLog, record)
          } else {
            this.addRecordToBuffer(record)
          }
        })
        .catch(() => {
          console.log('Error when checking if file exists: ', rotatingLog)
        })
    } else {
      this.addRecordToBuffer(record)
    }
  },

  addRecordToBuffer: function(record: any) {
    if (this.recordBuffer.length <= 0 && !this.initOnlyOnce) {
      // This logic MUST ensure that this.init() is only invoked ONCE
      this.init()
    }
    this.recordBuffer.push(record)
  },

  writeToLog: function(rotatingLog: string, record: any) {
    const recordAsString = JSON.stringify(record)

    if (recordAsString.indexOf('%c prev state') === -1) {
      writeToVcxLog(
        'ConnectMe.ReactNative',
        record.levelName,
        recordAsString,
        rotatingLog
      )
        .then(() => {
          //console.log('Wrote log message ' + recordAsString + ' to log file: ', rotatingLog)
        })
        .catch(error => {
          console.error(
            'Error writing log message ' + recordAsString + ' to log file: ',
            rotatingLog
          )
        })
    }
  },

  encryptLogFile: async function() {
    const rotatingLog = this.getVcxLogFile()
    this.encryptedLogFile = await encryptVcxLog(
      rotatingLog,
      CUSTOM_LOG_UTILS.encryptionKey
    )
    //console.log('Setting encrypted vcx log file to: ', this.encryptedLogFile)
    return this.encryptedLogFile
  },

  clearRecords: function() {},

  getRecords: function() {},
}
