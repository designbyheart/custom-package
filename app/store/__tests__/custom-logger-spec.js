// @flow
import { customLogger } from '../custom-logger'

describe('Custom Logger', () => {
  it('Custom Logger should log to file', () => {
    customLogger.log('test log statement', ' asdfasdf ')
    //console.log(customLogger.calculateRotatingLog)
  })
})
