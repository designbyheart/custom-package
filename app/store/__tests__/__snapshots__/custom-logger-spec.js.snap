// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fn:PiiHiddenActionTransformer should hide PII in TOKEN_SENT_SUCCESS 1`] = `
Object {
  "payment": Object {
    "error": null,
    "status": "SUCCESS",
    "tokenAmount": "****",
  },
  "type": "TOKEN_SENT_SUCCESS",
}
`;

exports[`fn:PiiHiddenActionTransformer should hide PII in WALLET_ADDRESSES_REFRESHED 1`] = `
Object {
  "type": "WALLET_ADDRESSES_REFRESHED",
  "walletAddresses": Object {
    "data": "****",
    "error": null,
    "status": "SUCCESS",
  },
}
`;

exports[`fn:PiiHiddenActionTransformer should hide PII in WALLET_BALANCE_REFRESHED 1`] = `
Object {
  "type": "WALLET_BALANCE_REFRESHED",
  "walletBalance": Object {
    "data": "****",
    "error": null,
    "status": "SUCCESS",
  },
}
`;

exports[`fn:PiiHiddenActionTransformer should hide PII in WALLET_HISTORY_REFRESHED 1`] = `
Object {
  "type": "WALLET_HISTORY_REFRESHED",
  "walletHistory": Object {
    "error": null,
    "status": "SUCCESS",
    "transactions": "****",
  },
}
`;

exports[`fn:PiiHiddenActionTransformer should hide data Passphrase from action logging 1`] = `
Object {
  "passphrase": "****",
  "status": "GENERATE_PHRASE_SUCCESS",
  "type": "GENERATE_RECOVERY_PHRASE_SUCCESS",
}
`;

exports[`fn:PiiHiddenActionTransformer should hide senderDID and pairwiseDID in MAP_CLAIM_TO_SENDER 1`] = `
Object {
  "claimUuid": "claimUUID",
  "logoUrl": "logoUrl",
  "myPairwiseDID": "****",
  "senderDID": "****",
  "type": "MAP_CLAIM_TO_SENDER",
}
`;

exports[`fn:PiiHiddenTransformer should hide all PII information with **** 1`] = `
Object {
  "backup": Object {
    "backupWalletPath": "/Users/CoreSimulator/Devices/5F89AA14-B694-465F-82C3-838552AB4B85/data/Containers/Data/Application/4536A441-4B70-4A30-8DCC-5274E6781535/Documents/backup.zip",
    "cloudBackupPending": false,
    "error": null,
    "lastSuccessfulBackup": "2018-06-27T18:27:35+05:30",
    "lastSuccessfulCloudBackup": "2018-06-27T18:27:35+05:30",
    "passphrase": "****",
    "prepareBackupStatus": "PREPARE_BACKUP_SUCCESS",
    "showBanner": true,
    "status": "BACKUP_COMPLETE",
  },
  "claim": "****",
  "claimOffer": "****",
  "config": Object {
    "agencyDID": "tjVxL8raUsG5s5ZzGhYV1",
    "agencyUrl": "https://agency-sandbox.evernym.com",
    "agencyVerificationKey": "VCNhKASjLU5tVWnZpjcyEsoV1QC3adrPVMCCjeKMobn",
    "isAlreadyInstalled": true,
    "isHydrated": false,
    "isInitialized": false,
    "paymentMethod": "sov",
    "poolConfig": "{\\"data\\":{\\"alias\\":\\"Node1\\",\\"blskey\\":\\"4N8aUNHSgjQVgkpm8nhNEfDf6txHznoYREg9kirmJrkivgL4oSEimFF6nsQ6M41QvhM2Z33nves5vfSn9n1UwNFJBYtWVnHYMATn76vLuL3zU88KyeAYcHfsih3He6UHcXDxcaecHVz6jhCYz1P2UZn2bDVruL5wXpehgBfBaLKm3Ba\\",\\"client_ip\\":\\"35.164.240.131\\",\\"client_port\\":9702,\\"node_ip\\":\\"35.164.240.131\\",\\"node_port\\":9701,\\"services\\":[\\"VALIDATOR\\"]},\\"dest\\":\\"Gw6pDLhcBcoQesN72qfotTgFa7cbuqZpkX3Xo6pLhPhv\\",\\"identifier\\":\\"Th7MpTaRZVRYnPiabds81Y\\",\\"txnId\\":\\"fea82e10e894419fe2bea7d96296a6d46f50f93f9eeda954ec461b2ed2950b62\\",\\"type\\":\\"0\\"}
{\\"data\\":{\\"alias\\":\\"Node2\\",\\"blskey\\":\\"37rAPpXVoxzKhz7d9gkUe52XuXryuLXoM6P6LbWDB7LSbG62Lsb33sfG7zqS8TK1MXwuCHj1FKNzVpsnafmqLG1vXN88rt38mNFs9TENzm4QHdBzsvCuoBnPH7rpYYDo9DZNJePaDvRvqJKByCabubJz3XXKbEeshzpz4Ma5QYpJqjk\\",\\"client_ip\\":\\"35.164.240.131\\",\\"client_port\\":9704,\\"node_ip\\":\\"35.164.240.131\\",\\"node_port\\":9703,\\"services\\":[\\"VALIDATOR\\"]},\\"dest\\":\\"8ECVSk179mjsjKRLWiQtssMLgp6EPhWXtaYyStWPSGAb\\",\\"identifier\\":\\"EbP4aYNeTHL6q385GuVpRV\\",\\"txnId\\":\\"1ac8aece2a18ced660fef8694b61aac3af08ba875ce3026a160acbc3a3af35fc\\",\\"type\\":\\"0\\"}
{\\"data\\":{\\"alias\\":\\"Node3\\",\\"blskey\\":\\"3WFpdbg7C5cnLYZwFZevJqhubkFALBfCBBok15GdrKMUhUjGsk3jV6QKj6MZgEubF7oqCafxNdkm7eswgA4sdKTRc82tLGzZBd6vNqU8dupzup6uYUf32KTHTPQbuUM8Yk4QFXjEf2Usu2TJcNkdgpyeUSX42u5LqdDDpNSWUK5deC5\\",\\"client_ip\\":\\"35.164.240.131\\",\\"client_port\\":9706,\\"node_ip\\":\\"35.164.240.131\\",\\"node_port\\":9705,\\"services\\":[\\"VALIDATOR\\"]},\\"dest\\":\\"DKVxG2fXXTU8yT5N7hGEbXB3dfdAnYv1JczDUHpmDxya\\",\\"identifier\\":\\"4cU41vWW82ArfxJxHkzXPG\\",\\"txnId\\":\\"7e9f355dffa78ed24668f0e0e369fd8c224076571c51e2ea8be5f26479edebe4\\",\\"type\\":\\"0\\"}
{\\"data\\":{\\"alias\\":\\"Node4\\",\\"blskey\\":\\"2zN3bHM1m4rLz54MJHYSwvqzPchYp8jkHswveCLAEJVcX6Mm1wHQD1SkPYMzUDTZvWvhuE6VNAkK3KxVeEmsanSmvjVkReDeBEMxeDaayjcZjFGPydyey1qxBHmTvAnBKoPydvuTAqx5f7YNNRAdeLmUi99gERUU7TD8KfAa6MpQ9bw\\",\\"client_ip\\":\\"35.164.240.131\\",\\"client_port\\":9708,\\"node_ip\\":\\"35.164.240.131\\",\\"node_port\\":9707,\\"services\\":[\\"VALIDATOR\\"]},\\"dest\\":\\"4PS3EDQ3dW1tci1Bp6543CfuuebjFrg36kLAUcskGfaA\\",\\"identifier\\":\\"TWwCRQRZ2ZHMJFn9TzLp7W\\",\\"txnId\\":\\"aa5e817d7cc626170eca175822029339a444eb0ee8f0bd20d3b0b76e566fb008\\",\\"type\\":\\"0\\"}",
    "showErrorAlerts": false,
    "vcxInitializationError": null,
    "vcxInitializationState": "VCX_INIT_SUCCESS",
  },
  "connections": Object {
    "connectionThemes": Object {
      "default": Object {
        "primary": "rgba(119, 119, 119, 1.0)",
        "secondary": "rgba(119, 119, 119, 0.6)",
      },
    },
    "data": "****",
  },
  "deepLink": Object {
    "error": undefined,
    "isLoading": true,
    "tokens": "****",
  },
  "eula": Object {
    "isEulaAccept": true,
  },
  "history": Object {
    "data": "****",
    "isLoading": true,
  },
  "invitation": Object {},
  "ledger": Object {
    "fees": Object {
      "data": Object {
        "refreshTime": "2018-01-10T04:32:43+05:30",
        "transfer": "0.00001",
      },
      "error": null,
      "status": "SUCCESS",
    },
  },
  "lock": Object {
    "checkPinStatus": "CHECK_PIN_IDLE",
    "inRecovery": "false",
    "isAppLocked": false,
    "isLockEnabled": "true",
    "isTouchIdEnabled": true,
    "pendingRedirection": undefined,
    "showDevMode": false,
  },
  "offline": Object {
    "offline": false,
  },
  "onfido": Object {
    "applicantId": null,
    "error": null,
    "onfidoConnectionStatus": "IDLE",
    "onfidoDid": null,
    "status": "IDLE",
  },
  "proof": "****",
  "proofRequest": "****",
  "pushNotification": Object {
    "error": undefined,
    "isAllowed": true,
    "isFetching": false,
    "isPristine": true,
    "navigateRoute": undefined,
    "notification": undefined,
    "pendingFetchAdditionalDataKey": undefined,
    "pushToken": undefined,
  },
  "question": Object {
    "data": "****",
    "storageStatus": "RESTORE_SUCCESS",
  },
  "restore": Object {
    "error": null,
    "restoreFile": Object {
      "fileName": "backup.zip",
      "fileSize": 22,
      "type": "application/zip",
      "uri": "file://data/Containers/Data/Application/021D78D1-044B-4874-BE8D-EC71781CA978/tmp/com.evernym.connectme.callcenter-Inbox/backup.zip",
    },
    "status": "none",
  },
  "route": Object {
    "currentScreen": "TabQRCodeScanner",
    "timeStamp": 1557756720914,
  },
  "sendlogs": Object {
    "encryptLogStatus": false,
  },
  "smsPendingInvitation": Object {},
  "txnAuthorAgreement": Object {
    "haveAlreadySignedAgreement": false,
    "status": "IDLE",
    "taaAcceptedVersion": "",
    "text": "",
    "thereIsANewAgreement": false,
    "version": "",
  },
  "user": Object {
    "avatarName": "user-avatar.jpeg",
    "error": null,
    "isFetching": false,
    "userOneTimeInfo": "****",
  },
  "wallet": Object {
    "backup": Object {
      "backupPath": "test_backup_path",
      "encryptionKey": "walletEncryptionKey",
      "error": null,
      "fileStatus": "IDLE",
      "latest": "2017-09-06T00:00:00+05:30",
      "status": "IDLE",
    },
    "payment": Object {
      "error": null,
      "status": "SUCCESS",
      "tokenAmount": "****",
    },
    "walletAddresses": Object {
      "data": "****",
      "error": null,
      "status": "SUCCESS",
    },
    "walletBalance": Object {
      "data": "****",
      "error": null,
      "status": "SUCCESS",
    },
    "walletHistory": Object {
      "error": null,
      "status": "SUCCESS",
      "transactions": "****",
    },
  },
}
`;
